### PR TITLE
#3930 migrate shared-utils to eslint

### PR DESCRIPTION
**Pull request by desfero**

### CODE REVIEW

File: packages/eslint-config/base.js [hunk 1]
```diff
@@ -122,8 +122,8 @@ module.exports = {
     "@typescript-eslint/no-magic-numbers": [
       "error",
       {
-        // 0 is often used to access first element of an array
-        ignore: [0],
+        // ignore some common numbers
```
Comment (arjunu) [inline packages/eslint-config/base.js:125]: there is `ignoreArrayIndexes` option for this rule, should we use?

File: packages/shared-utils/src/utils/StringUtils.ts [hunk 2]
```diff
@@ -9,6 +10,7 @@ export function isExternalUrl(s: string): boolean {
 
 export function isEmail(s: string): boolean {
   // General Email Regex (RFC 5322 Official Standard)
+  // eslint-disable-next-line no-control-regex
```
Comment (arjunu) [inline packages/shared-utils/src/utils/StringUtils.ts:13]: can this be fixed with `new RegExp(...)`?

File: packages/shared-modules/src/tests/auth/createLightWalletWithKeyPair.ts [hunk 3]
```diff
@@ -22,10 +22,10 @@ export const createLightWalletWithKeyPair = async (
     salt,
   });
   // create keypair
-  const keyFromPassword = promisify<any>(
+  const keyFromPassword = promisify<[string], any>(
     lightWalletInstance.keyFromPassword.bind(lightWalletInstance),
   );
-  const walletKey: any = await keyFromPassword(password);
+  const walletKey = await keyFromPassword(password);
```
Comment (Moejoe90) [inline packages/shared-modules/src/tests/auth/createLightWalletWithKeyPair.ts:28]: :+1:

File: packages/shared-utils/src/opaque-types/types.ts [hunk 4]
```diff
@@ -3,4 +3,4 @@
  * see: https://codemix.com/opaque-types-in-javascript/
  */
 export type Opaque<K, T> = T & { __TYPE__: K };
-export type OpaqueType<O extends Opaque<any, any>> = O["__TYPE__"];
+export type OpaqueType<O extends Opaque<unknown, unknown>> = O["__TYPE__"];
```
Comment (Moejoe90) [inline packages/shared-utils/src/opaque-types/types.ts:6]: :+1:

**Review by Moejoe90 — APPROVED** [review]
looks good to me

File: packages/eslint-config/base.js (hunk 1 above)
Comment (desfero) [reply packages/eslint-config/base.js:125]: it won't fix the problem fully as 0 is also used often outside of accessing arrays AFAIK

File: packages/shared-utils/src/utils/StringUtils.ts (hunk 2 above)
Comment (desfero) [reply packages/shared-utils/src/utils/StringUtils.ts:13]: Not sure really as this rule checks some kind of control characters that we use in regex for email both `/regex/` and `new RegExp` gonna fail

**Review by arjunu — APPROVED** [review]

**Pull request merged**